client/logmon/logmon.go

```diff
@@ -0,0 +1,257 @@
+package logmon
+
+import (
+	"fmt"
+	"io"
+	"os/exec"
+	"strings"
+	"time"
+
+	hclog "github.com/hashicorp/go-hclog"
+	plugin "github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/nomad/client/driver/logging"
+	"github.com/hashicorp/nomad/client/lib/fifo"
+	"github.com/hashicorp/nomad/helper/discover"
+)
+
+const (
+	// processOutputCloseTolerance is the length of time we will wait for the
+	// launched process to close its stdout/stderr before we force close it. If
+	// data is written after this tolerance, we will not capture it.
+	processOutputCloseTolerance = 2 * time.Second
+)
+
+var Handshake = plugin.HandshakeConfig{
+	// This isn't required when using VersionedPlugins
+	ProtocolVersion:  1,
+	MagicCookieKey:   "LOGMON_PLUGIN",
+	MagicCookieValue: "logmon",
+}
+
+type LogConfig struct {
+	// LogDir is the host path where logs are to be written to
+	LogDir string
+
+	// StdoutLogFile is the path relative to LogDir for stdout logging
+	StdoutLogFile string
+
+	// StderrLogFile is the path relative to LogDir for stderr logging
+	StderrLogFile string
+
+	// StdoutFifo is the path on the host to the stdout pipe
+	StdoutFifo string
+
+	// StderrFifo is the path on the host to the stderr pipe
+	StderrFifo string
+
+	// MaxFiles is the max rotated files allowed
+	MaxFiles int
+
+	// MaxFileSizeMB is the max log file size in MB allowed before rotation occures
+	MaxFileSizeMB int
+
+	// UID is id for the desired user to write log files as
+	UID int
+
+	// GID is id for the desried group to write log files as
+	GID int
+}
+
+type LogMon interface {
+	Start(*LogConfig) error
+	Stop() error
+}
+
+func LaunchLogMon(logger hclog.Logger) (LogMon, *plugin.Client, error) {
```

Comment:
I think this is fine for now but we should probably add it to the plugin-launcher eventually